cluster/cluster.go

```diff
@@ -32,13 +33,34 @@ func Join(ctx context.Context, cfg Config) (*Cluster, error) {
 		zap.ReplaceGlobals(logger)
 	}
 
+	if cfg.etcdConfig.ClusterState == embed.ClusterStateFlagNew || len(clientUrls) == 0 {
+		clientUrls = urlsToString(cfg.etcdConfig.LCUrls)
+	} else if cfg.etcdConfig.ClusterState == embed.ClusterStateFlagExisting && len(clientUrls) == 0 {
+		return nil, fmt.Errorf("joining an existing cluster requires at least one client url from a member from the existing cluster")
+	}
+
+	clientCfg := clientv3.Config{
+		Endpoints:   clientUrls,
+		DialTimeout: 5 * time.Second,
+	}
+	client, err := clientv3.New(clientCfg)
```

Comment:
this is a different behaviour because previously the client field referred to the local etcd. I think we would still want to preserve that behaviour for `MemberAdd` and `MemberList` calls.  

We would only need to create remote etcd client connections for the case of a `exisiting` cluster for the Join call right?

Reply:
Before when we were testing it, we called `MemberAdd` on the `cluster` struct, so it was `c.MemberAdd`, and the client we called it on had the initial node clientUrl. We can't call MemberAdd on a `clientv3.Client` made from its own local clientUrl since it's not part of the existing cluster. We might be able to create the `clientv3.Client` like above then remake one with the new local `clientUrl` before returning, but we still need the non local etcd `clientv3.Client` to call `MemberAdd`

Reply:
> We would only need to create remote etcd client connections for the case of a `exisiting` cluster for the Join call right?

Yeah, thats why I added the case where it just takes the `Config`'s clientUrl if its a new cluster + if the `clientUrls` passed in is empty

Reply:
I understand that we need to create the etcd clients for the initial cluster in order to do the call the member add on a remote etcd in the case where the cluster is existing. 

What I'm suggesting is to then use the local etcd as a client for the registry that is initalized later in the function. I was initially purposing the same for MemberList and MemberAdd but I don't think those matter as much.